src/router.rs

```diff
@@ -61,6 +61,11 @@ impl<State: 'static> Router<State> {
             // if not then fallback to the behavior of HTTP GET else proceed as usual
 
             self.route(path, http::Method::GET)
+        } else if self.method_map.values().any(|r| r.recognize(path).is_ok()) {
```

Comment:
This code contains quite a few branches which can be hard to follow; could you maybe add a comment here explaining what this does?